Add GDAL_DCAP_FEATURE_STYLES capability

nyalldawson opened the pull request:
To indicate when a driver has support for feature styles.

Note that I've added this capability for many existing drivers, based on grepping the source for use of SetStyleString. It's possible the capability should not be present for some of these - e.g. I don't think WFS should indicate this capability?

rouault commented:
There are other drivers missing like ogr/ogrsf_frmts/mitab , which call GetStyleString() / m_pszStyleString
Regarding WFS, the support for feature style is rather hypothetical, in the case where you would use a unusual output format (like MapInfo), that would support itself feature style.

nyalldawson commented:
Ok, I've removed WFS and added the other drivers.

What I'm debating is whether we should have two different flags here... one for write support, and one for read support. What do you think?

rouault commented:
> What I'm debating is whether we should have two different flags here... one for write support, and one for read support. What do you think?

I was wondering about that too, but I believe that all drivers that support styling in read and have write support also support writing styling

The SQLite driver shouldn't declare the capability. Its use of ogr feature style is purely internal to implement the SQL SQLite dialect.

nyalldawson commented:
> The SQLite driver shouldn't declare the capability. Its use of ogr feature style is purely internal to implement the SQL SQLite dialect.

ok, fixed.

rouault commented:
Committed in trunk r40706. Thanks
